src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbol.cs

```diff
@@ -33,6 +33,27 @@ protected static void ReportBadRefToken(TypeSyntax returnTypeSyntax, DiagnosticB
             }
         }
 
+        protected bool AreContainingSymbolLocalsZeroed
```

Comment:
You could do 
```C#
protected bool AreContainingSymbolLocalsZeroed => ContainingSymbol switch
{
    SourceMethodSymbol m => m.AreLocalsZeroed,
    SourceNamedTypeSymbol t => t.AreLocalsZeroed,
    _ => true
};
```

Reply:
Does this handle the scenario where the ContainingSymbol is a SourceFieldSymbolFromDeclarator and the containing type has SkipLocalsInitAttribute?

Reply:
I think when I was triaging this I was surprised to find this as the ContainingSymbol for some scenario or other, but I tried to create a test that would create a bad behavior here and couldn't do it, so I'm good with taking your suggestion.

Reply:
Aleksey came up with a test case involving a field initializer containing a lambda, where the ContainingSymbol is a FieldSymbol and we have to instead look up to the ContainingType. So I reverted the suggestion.

Reply:
Were you thinking of a lambda inside a field initializer?

Reply:
I could see that as being a reasonable scenario that would be broken with my code.